engines/director/util.cpp

```diff
@@ -760,6 +750,22 @@ Common::Path resolvePath(const Common::String &path, const Common::Path &base, b
 	return Common::Path();
 }
 
+Common::Path resolvePath(const Common::String &path, const Common::Path &base, bool directory, const char **exts) {
```

Comment:
What is the test case for this change? E.g. what is the directory structure and the requested file path, so this now works?

I also, think, given the complexity of this code, it would be nice to eventually add a test suite, e.g. simulate a filesystem (with Common::Archive) and then query it and compare the results. Otherwise, I am afraid, this code is way too fragile.

Reply:
As I mentioned in the commit message, the test case for this change is accessing a movie only available through SearchMan and not the filesystem (e.g. inside a ProjectorArchive), where the Lingo code references `go(1, "A")` and the filename in SearchMan is `F:\POPUP\A.DIR`. Previously we would only test for different file extensions at the filesystem level, and not the various caches.